app/models/auth_service_account.rb

```diff
@@ -0,0 +1,6 @@
+class AuthServiceAccount < Authentication
```

Comment:
Or why not just use AuthUseridPassword?  Is there something significantly different about a service account?  Also would one ever have a "regular" userid/pass *and* a service account on a provider?